app/Services/GitHub/Contracts/GitHubClientFactoryContract.php

```
@@ -2,9 +2,7 @@
 
 namespace App\Services\GitHub\Contracts;
 
-use Github\Client;
-
 interface GitHubClientFactoryContract
 {
-    public function make(?string $connection = null, bool $withCache = true): Client;
+    public function make(?string $connection = null, bool $withCache = true): GitHubClientContract;
```

PHPMD (2.15.0)

The method make has a boolean flag argument $withCache, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)